src/main/java/ch/njol/skript/effects/base/SetEffect.java

```diff
@@ -0,0 +1,147 @@
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+/**
+ * Contains the default effects of Skript.
+ * 
+ * @author Peter Güttinger
+ */
+package ch.njol.skript.effects.base;
+
+import java.util.Arrays;
+import java.util.function.BiConsumer;
+
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.SkriptAPIException;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+
+/**
+ * A class that is to be used for boolean expressions.
+ * <p>
+ * A boolean expression should be a condition for getting it's value.
+ */
+public abstract class SetEffect<T> extends Effect {
+
+	private Expression<Boolean> value;
+	private Expression<T> expression;
+	private boolean make, negated;
+
+	/**
+	 * Registers an effect with patterns "set property of %type% to %boolean%" and "set %types%'[s] property to %boolean%"
+	 * 
+	 * @param effect The SetEffect class that you are registering.
+	 * @param property The property of the syntax.
+	 * @param type The type classinfo of the syntax. Can be plural.
+	 */
+	public static void register(Class<? extends SetEffect<?>> effect, String property, String type) {
+		Skript.registerEffect(effect, "set " + property + " of %" + type + "% to %boolean%",
+				"set %" + type + "%'[s] " + property + " to %boolean%");
+	}
+	
+	/**
+	 * Registers an effect with patterns "set property of %type% to %boolean%", "set %types%'[s] property to %boolean%"
+	 * and "make %types% makeProperty" with "force %types% to makeProperty"
+	 * 
+	 * @param effect The SetEffect class that you are registering.
+	 * @param property The property of the syntax.
+	 * @param makeProperty The property of the syntax used for "make %types% makeProperty".
+	 * @param type The type classinfo of the syntax. Can be plural.
+	 */
+	public static void registerMake(Class<? extends SetEffect<?>> effect, String property, String makeProperty, String type) {
+		Skript.registerEffect(effect, "set " + property + " of %" + type + "% to %boolean%",
+				"set %" + type + "%'[s] " + property + " to %boolean%",
+				"make %" + type + "% " + makeProperty,
+				"force %" + type + "% to " + makeProperty,
+				"make %" + type + "% not " + makeProperty,
+				"force %" + type + "% to not " + makeProperty);
```

Comment:
maybe allow `not to` too as in `force {_item} not to be all`?

Reply:
That's not proper

Reply:
It's `to not be` like in Shakespeare

Reply:
isn't the line literally "to be, or **not to be**, that is the question"?

Reply:
Ya that's what exists already. `force %type% to not`

Reply:
that isn't `not to be` though?
``force %type% to not [be all]`` is what exists, which isn't `force %type% not to [be all]`
both are valid, as far as I know

Reply:
I had a long old idea to add `be` as an optional noun.

Reply:
yeah, I get that.
I'm only talking about ``not to`` being adding as a syntax for this pull, what Walrus suggested.
the `be` would indeed *be* optional, it'd just allow two ways to construct the negated `force` syntax.

Reply:
`force player's name not to display` isn't proper where as `force the player's shield to not display` is correct.

Reply:
Both are correct: <https://www.britannica.com/dictionary/eb/qa/to-not-be-or-not-to-be>

Reply:
That states to add context typically with `not be`

Reply:
Lime, I think this is missing the point, and this isn't really worth the effort of arguing over imo.
Let me just sum up my thoughts and I'll leave it here.
First, both constructions are grammatically correct:
> But both examples are grammatically correct.

Second, "not to be" is more common than "to not be":
> However, the form that does not "split" the infinitive is much more common in print, and using this form (not to be; to go boldly; not to take) will work in almost every circumstance. 

It's fine if we decide to only keep `to not be` as the only construction in Skript, but I think we'd be better served adding `not to be` as well, as it's the more common usage (and personally, sounds more pleasing to my ear).